File: app/routers/account.py
"""Account deletion endpoint (multi-tenant).

`DELETE /api/account` erases the authenticated user's app data (Postgres rows +
Neo4j subgraph). The client follows up with the `delete_account` Supabase RPC to
remove the auth identity, then signs out. `user_id` comes from the auth
dependency, so a user can only ever delete themselves.
"""
from uuid import UUID

from fastapi import APIRouter, Depends

from ..account import delete_account
from ..auth import get_current_user_id


router = APIRouter(prefix="/api/account", tags=["account"])


@router.delete("")
async def delete_account_endpoint(user_id: UUID = Depends(get_current_user_id)):
    return delete_account(user_id)

File: mobile/lib/account.ts
import { apiFetch } from './api';
import { supabase } from './supabase';

// Permanently delete the account. Two steps, in order:
//   1. Backend wipes all app data (Postgres rows + the Neo4j subgraph).
//   2. A SECURITY DEFINER Supabase RPC removes the auth identity itself — the
//      backend never holds the service-role key, so this self-delete lives in
//      Postgres and runs as the authenticated user (auth.uid()).
// Returns true once the auth user is gone; the caller then signs out.
export async function deleteAccount(): Promise<boolean> {
  try {
    const res = await apiFetch('/api/account', { method: 'DELETE' });
    if (!res.ok) return false;
    const { error } = await supabase.rpc('delete_account');
    return !error;
  } catch {
    return false;
  }
}
